be/src/connector/connector.h

```diff
@@ -8,56 +8,80 @@
 
 #include "gen_cpp/PlanNodes_types.h"
 #include "runtime/runtime_state.h"
+
 namespace starrocks {
 
+class ExprContext;
 namespace vectorized {
 class ConnectorScanNode;
-// class ConnectorSinkNode;
+class RuntimeFilterProbeCollector;
 } // namespace vectorized
 
 namespace connector {
 
 class DataSource {
 public:
-    virtual Status open(RuntimeState* state);
-    virtual void close(RuntimeState* state);
-    virtual Status get_next(RuntimeState* state, vectorized::ChunkPtr* chunk);
-
-    virtual Status do_open(RuntimeState* state) = 0;
-    virtual void do_close(RuntimeState* state) = 0;
-    virtual Status do_get_next(RuntimeState* state, vectorized::ChunkPtr* chunk) = 0;
+    virtual ~DataSource() = default;
+    virtual Status init() { return Status::OK(); }
```

Comment:
do we need `init` function? because the initialization is invoked by `DataSourceProvider`

Reply:
Because there are some methods like 

```
    void set_runtime_profile(RuntimeProfile* runtime_profile) { _runtime_profile = runtime_profile; }
    void set_predicates(const std::vector<ExprContext*>& predicates) { _conjunct_ctxs = predicates; }
    void set_runtime_filters(const vectorized::RuntimeFilterProbeCollector* runtime_filters) {
        _runtime_filters = runtime_filters;
    }
    void set_read_limit(const uint64_t limit) { _read_limit = limit; }
```

I hope there is a method can be called after settings those fields.

Reply:
The steps are:
1. created by data source provider.
2. set some query context like `runtime filter` and `predicates`
3. init

Reply:
remove `init` method